dask/highlevelgraph.py

```diff
@@ -457,6 +478,67 @@ def cull(self, keys: Set):
 
         return HighLevelGraph(ret_layers, ret_dependencies, ret_key_deps)
 
+    def map_basic_layers(
+        self, func: Callable[[BasicLayer], Mapping]
+    ) -> "HighLevelGraph":
+        """Map `func` on each basic layer and returns a new high level graph.
+
+        `func` should take a BasicLayer as input and return a new Mapping as output
+        and **cannot** change the dependencies between Layers.
+
+        If `func` returns a non-Layer type, it will be wrapped in a `BasicLayer`
+        object automatically.
+
+        Parameters
+        ----------
+        func : callable
+            The function to call on each BasicLayer
+
+        Returns
+        -------
+        hlg : HighLevelGraph
+            A high level graph containing the transformed BasicLayers and the other
+            Layers untouched
+        """
+        layers = {}
+        for k, v in self.layers.items():
+            if isinstance(v, BasicLayer):
+                layer = func(v)
+                if not isinstance(layer, Layer):
+                    layer = BasicLayer(layer)
+                layers[k] = layer
+            else:
+                layers[k] = v
```

Comment:
Since `HighLevelGraph.__init__` will cast layers which are not `Layer` types to a `BasicLayer`, this can be simplified to something like

```suggestion
        layers = {
            k: func(v) if isinstance(v, BasicLayer) else v
            for k, v in self.layers.items()
        }
```

Reply:
@jrbourbeau due to long delays because of timezones, and the impressive backlog that @madsbk has built up, I would encourage you to push directly if you have high confidence in these things (assuming that @madsbk is ok with this)

Reply:
That is fine with me!

Reply:
... to both comments :)

Reply:
Great, will do!